device.go

```diff
@@ -25,10 +25,11 @@ import (
 // DeviceDesc is a representation of a USB device descriptor.
 type DeviceDesc struct {
 	// Bus information
-	Bus     int   // The bus on which the device was detected
-	Address int   // The address of the device on the bus
-	Speed   Speed // The negotiated operating speed for the device
-	Port    int   // The usb port on which the device was detected
+	Bus     int     // The bus on which the device was detected
+	Address int     // The address of the device on the bus
+	Speed   Speed   // The negotiated operating speed for the device
+	Port    int     // The usb port on which the device was detected
+	Path    []uint8 //The usb port path of the device
```

Comment:
First, make this a path of `int`, not `uint8`. Go uses int as the default integer type and there's no reason to present `uint8` specifically here. That way it will also match the type of "Port".

Second, this is an insufficient description, I think. At very least you need to mention the ordering (from the root of the bus), mention how the root behaves (root hub has the port number 0 and an empty path) and that the combination of Bus+Path is a unique identifier for a physical device.

Reply:
I think defining a DevicePath type might help in documenting this better. I'll pop it in and see what you think.